app.rb

```diff
@@ -1,92 +1,36 @@
-require_relative 'person'
-require_relative 'book'
-require_relative 'student'
-require_relative 'teacher'
-require_relative 'rental'
+require_relative 'create_people'
+require_relative 'create_book'
+require_relative 'create_rental'
 
 class App
-  attr_accessor :people, :books, :rentals
-
   def initialize
-    @people = []
-    @books = []
     @rentals = []
+    @rental = CreateRental.new
+    @book = CreateBook.new
+    @person = CreatePeople.new
   end
 
-  def list_books
-    puts 'No books' if @books.empty?
-    @books.each do |book|
-      puts "ID: #{book.id} Title: #{book.title}, Author: #{book.author}"
-    end
-  end
-
-  def list_people
-    puts 'No Person' if @people.empty?
-    @people.each do |person|
-      puts "[#{person.class}]: ID: #{person.id}, Name: #{person.name}"
-    end
-  end
-
-  def ask_student
-    print 'name: '
-    name = gets.chomp
-
-    print 'age: '
-    age = gets.chomp.to_i
-
-    print 'Permission: '
-    parent_permission = gets.chomp
-
-    @people << Student.new(name, age, parent_permission: parent_permission)
-  end
-
-  def ask_teacher
-    print 'name: '
-    name = gets.chomp
-
-    print 'age: '
-    age = gets.chomp.to_i
-
-    print 'specialization: '
-    specialization = gets.chomp
-
-    @people << Teacher.new(age, specialization, name, parent_permission: true)
-  end
-
-  # create preson
-  def create_person
-    puts 'What kind of person do you want to create? (for student type 1 and for teacher type 2)'
-    person_type = gets.chomp.to_i
-
-    case person_type
+  def process_input(input, app)
+    case input
     when 1
-      ask_student
+      app.list_books
     when 2
-      ask_teacher
-    else
-      puts 'Invalid person'
-      return
+      app.list_people
+    when 3
+      app.create_person
+    when 4
+      app.create_book
+    when 5
+      app.create_rental
+    when 6
+      app.list_rental
     end
-
-    puts 'Person created successfully'
-  end
-
-  def create_book
-    print 'Title: '
-    title = gets.chomp
-
-    print 'Author: '
-    author = gets.chomp
-
-    @books << Book.new(title, author)
-
-    puts 'Book created successfully'
   end
 
   def create_rental
```

Comment:
- [ ] Your code is gradually taking shape, however, most of the methods are still playing multiple roles, which we want to eliminate in key methods. Endeavor to refactor all requests, for instance `get.chomps`  to be handled in different class/ methods. We can then reference those values which were stored in objects during the request in our actual methods.